src/visitors/defuse_analyze_visitor.hpp

```diff
@@ -50,6 +50,9 @@ enum class DUState {
     NONE
 };
 
+/// Variable type processed by DefUseAnalyzeVisitor
+enum class DUVariableType { Local, Global };
+
```

Comment:
is this necessary since it could be already specified with LU, LD

Reply:
EDIT: in this case maybe we could simplify, rework `update_defuse_chain` to use this new enum. Dunno

Reply:
Just to clarify, this enum is used to define whether we are looking for the `L`, `U` or `CD` states of variable if it's a global variable in the `SymTab` or `LU`, `LD` or `CD` if it's a local variable in the `SymTab`. I should add a comment about this here